### PR TITLE
API and Component Buildout for Homepage

### PR DESCRIPTION
**Pull request by sreya-kata**
Did the following:
- reorganized component file structure so every utilized component has its own folder with a js file and (future) css file
- modified homepage content model in CMS to coincide with new design
- data fetched all CMS components of the homepage into a graphql query
- made initial components for all distinct homepage components based on hi-fis (hero, mission, inclusivity w/carousel, stay in the know, and final carousel)

Note: from the code, "mission", "inclusivity", and "stay in the know" seem similar but separate react components are made as they are each paired with another feature (ie: inclusivity has a passive carousel, "stay in the know" has a featured image, the stylings slightly differ). See hi-fi representation below:
hifi

### CODE REVIEW

File: src/components/homepage/carousel/carousel.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import React from "react"
+import Img from "gatsby-image"
+
+const Carousel = ({ title, carouselItems }) => {
+  return (
+    <>
+      <p>{title}</p>
+      {carouselItems.map(item => {
+        return (
+          <div>
```
Comment (josephmannis) [inline src/components/homepage/carousel/carousel.js]: 🔑

File: src/components/homepage/carousel/carousel.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+            <h4>{item.title}</h4>
+            <p>{item.subTitle}</p>
+            <a href={item.link}>Learn More</a>
```
Comment (josephmannis) [inline src/components/homepage/carousel/carousel.js]: Can you please add target="_blank"

File: src/components/homepage/inclusivity/inclusivity.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import React from "react"
+import Img from "gatsby-image"
+
+const Inclusivity = ({ title, blurbText, ctaTitle, ctaLink, carousel }) => {
+  return (
+    <>
+      <h2>{title}</h2>
+      <p>{blurbText.blurbText}</p>
+      <a href={ctaLink}>
```
Comment (josephmannis) [inline src/components/homepage/inclusivity/inclusivity.js]: 🎯

File: src/components/homepage/inclusivity/inclusivity.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+        <button type="button">{ctaTitle}</button>
```
Comment (josephmannis) [inline src/components/homepage/inclusivity/inclusivity.js]: Can you please add an aria label here?

File: src/components/homepage/mission/mission.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import React from "react"
+
+const Mission = ({ title, blurbText, ctaTitle, ctaLink }) => {
+  return (
+    <>
+      <h2>{title}</h2>
+      <p>{blurbText.blurbText}</p>
+      <a href={ctaLink}>
```
Comment (josephmannis) [inline src/components/homepage/mission/mission.js]: 🎯 and aria label for the button, ty!

File: src/components/homepage/stay-in-the-know/stay-in-the-know.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+  blurbText,
+  ctaTitle,
+  ctaLink,
+  upcomingEvent,
+}) => {
+  return (
+    <>
+      <h2>{title}</h2>
+      <p>{blurbText.blurbText}</p>
+      <a href={ctaLink}>
```
Comment (josephmannis) [inline src/components/homepage/stay-in-the-know/stay-in-the-know.js]: Same here

File: src/components/layout/layout.js [hunk 7]
```diff
@@ -29,7 +29,6 @@ const Layout = ({ children }) => {
       <div
         style={{
           margin: `0 auto`,
-          maxWidth: 960,
```
Comment (josephmannis) [inline src/components/layout/layout.js:32]: Can you pull out the inline styles please?

File: src/pages/index.js [hunk 8]
```diff
@@ -1,47 +1,103 @@
...
+        carousel={homePage.everyoneWelcomeCarousel}
+      />
+      <StayInTheKnow
+        title={stayInTheKnow.title}
+        blurbText={stayInTheKnow.blurbText}
+        ctaTitle={stayInTheKnow.ctaTitle}
+        ctaLink={stayInTheKnow.ctaLink}
+        upcomingEvent={homePage.upcomingEvent}
+      />
+      <Carousel
```
Comment (josephmannis) [inline src/pages/index.js:46]: This is very neatly organized!

File: src/components/homepage/carousel/carousel.js (hunk 2 above)
Comment (josephmannis) [reply src/components/homepage/carousel/carousel.js]: Also, do these go to pages on the site or external links?

**Review by josephmannis — CHANGES_REQUESTED** [review]
Just a few things! Looks good otherwise

File: src/components/homepage/carousel/carousel.js (hunk 2 above)
Comment (sreya-kata) [reply src/components/homepage/carousel/carousel.js]: They're on the site. In the CMS, they're inputted as the name of page, but as the file name format (ie: "/get-involved")
Also I forgot to include the link in the query so adding that now^

**Comment by sreya-kata** [comment]
- Added key to carousel item by also adding a separate react component for a single carousel item
- Added target and rel attributes to links
- Added aria-labels to buttons

**Review by josephmannis — APPROVED** [review]
There's a warning in the console for an unused import. Can you please clean that up, then everything should be good!

**Pull request merged**